lionheart/models.py

```diff
@@ -200,34 +200,67 @@ class SoftDeleteQuerySet(models.query.QuerySet):
     SubClass of the standard Django QuerySet that ignores soft-deleted rows,
     and overwrites the delete function to update with soft-delete instead
     """
-    def delete(self):
-        return super(SoftDeleteQuerySet, self).update(deleted=SoftDeleteMixin.DELETED)
+    def delete(self, using='default', *args, **kwargs):
+        if not len(self):
+            return
+        for model in self:
+            model.delete(using, *args, **kwargs)
 
     def remove_permanently(self):
         return super(SoftDeleteQuerySet, self).delete()
 
     def deleted(self):
-        return super(SoftDeleteQuerySet, self).filter(deleted=SoftDeleteMixin.DELETED)
+        qs = super(SoftDeleteQuerySet, self).filter(deleted=SoftDeleteMixin.DELETED)
+        qs.__class__ = SoftDeleteQuerySet
```

Comment:
super(SoftDeleteQuerySet, self).filter will return a django.models.QuerySet, so any chained calls wont use SoftDeleteQuerySet Logic.  By setting the class to SoftDeleteQuerySet, chained calls will use the correct overridden functions